### PR TITLE
Add namespace parameter in service catalog

### LINKED ISSUE #2649: Unable to open addressspace console from provisioned service via service catalog
**Describe the bug**
As openshift user without cluster-admin rights I create namespace and run provision addressspace via service catalog. Addressspace is successfully created and I click on dashboard link in provisioned service but I get error. Also service broker pod still generates a lot of warning.

**Screenshots**
obrazek
obrazek
obrazek

**Additional context**
service broker
```
2019-04-14T08:16:11.305Z INFO  [Reflector] Resync
2019-04-14T08:16:11.306Z WARN  [Reflector] Exception doing resource update
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://172.30.0.1/apis/admin.enmasse.io/v1beta1/namespaces/myproject/consoleservices. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. consoleservices.admin.enmasse.io is forbidden: User "system:serviceaccount:myproject:service-broker" cannot list consoleservices.admin.enmasse.io in the namespace "myproject": no RBAC policy matched.
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:478) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:415) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:381) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:344) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:328) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(^CBaseOperation.java:165) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:607) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:67) ~[service-broker.jar:na]
	at io.enmasse.k8s.api.KubeCrdApi.list(KubeCrdApi.java:44) ~[service-broker.jar:na]
	at io.enmasse.k8s.api.cache.Reflector.resync(Reflector.java:78) ~[service-broker.jar:na]
	at io.enmasse.k8s.api.cache.Reflector.run(Reflector.java:53) ~[service-broker.jar:na]
	at io.enmasse.k8s.api.cache.Controller.lambda$start$0(Controller.java:22) ~[service-broker.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
2019-04-14T08:16:11.306Z INFO  [Reflector] Resync
2019-04-14T08:16:11.306Z WARN  [Reflector] Exception doing resource update
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://172.30.0.1/apis/admin.enmasse.io/v1beta1/namespaces/myproject/consoleservices. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. consoleservices.admin.enmasse.io is forbidden: User "system:serviceaccount:myproject:service-broker" cannot list consoleservices.admin.enmasse.io in the namespace "myproject": no RBAC policy matched.
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:478) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:415) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:381) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:344) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:328) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:165) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:607) ~[service-broker.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:67) ~[service-broker.jar:na]
	at io.enmasse.k8s.api.KubeCrdApi.list(KubeCrdApi.java:44) ~[service-broker.jar:na]
	at io.enmasse.k8s.api.cache.Reflector.resync(Reflector.java:78) ~[service-broker.jar:na]
	at io.enmasse.k8s.api.cache.Reflector.run(Reflector.java:53) ~[service-broker.jar:na]
	at io.enmasse.k8s.api.cache.Controller.lambda$start$0(Controller.java:22) ~[service-broker.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
```

### PR DESCRIPTION
**Pull request by lulf**
Add namespace parameter when creating address space using service
catalog, as the user does not have permissions to the address space
stored in the global enmasse namespace

Fixes #2649

### CODE REVIEW

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=ServiceCatalogWebTest

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=ServiceCatalogWebTest

File: service-broker/src/main/java/io/enmasse/osb/api/provision/OSBProvisioningService.java [hunk 1]
```diff
@@ -72,6 +72,7 @@ public Response provisionService(@Context SecurityContext securityContext,
         }
 
         String name = request.getParameter("name").orElse(service.getName() + "-" + shortenUuid(instanceId));
+        String namespace = request.getParameter("namespace").orElse(service.getName() + "-" + shortenUuid(instanceId));
```
Comment (k-wall) [inline service-broker/src/main/java/io/enmasse/osb/api/provision/OSBProvisioningService.java]: Why default the namespace name?  Wouldn't just rejecting the empty service be better?

Comment (lulf) [reply service-broker/src/main/java/io/enmasse/osb/api/provision/OSBProvisioningService.java]: Agreed, fixed this in latest commit

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=ServiceCatalogWebTest

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=ServiceCatalogWebTest

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=ServiceCatalogWebTest

File: systemtests/src/main/java/io/enmasse/systemtest/apiclients/ApiClient.java [hunk 2]
```diff
@@ -67,6 +67,11 @@ protected void reconnect() {
         connect();
     }
 
+    public void close() {
+        this.client.close();
```
Comment (k-wall) [inline systemtests/src/main/java/io/enmasse/systemtest/apiclients/ApiClient.java]: We'll NPE here if the client has not called connect(), which is likely to disguise a true failure. Please add a null check.

Comment (lulf) [reply systemtests/src/main/java/io/enmasse/systemtest/apiclients/ApiClient.java]: Fixed

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=ServiceCatalogWebTest

**Pull request merged**